mesonbuild/interpreter.py

```diff
@@ -2973,9 +2973,11 @@ def func_configure_file(self, node, args, kwargs):
             conffile = os.path.normpath(inputfile.relative_name())
             if conffile not in self.build_def_files:
                 self.build_def_files.append(conffile)
-        # Install file if requested
+        # Install file if requested, we check for the empty string
+        # for backwards compatibility. That was the behaviour before
+        # 0.45.1 so preserve it.
         idir = kwargs.get('install_dir', None)
-        if isinstance(idir, str):
+        if isinstance(idir, str) and idir != '':
```

Comment:
I am not sure if this is used throughout meson codebase but [PEP8](https://www.python.org/dev/peps/pep-0008/#id51) recommends `if isinstance(idir, str) and idir:`